pkg/framer/property_interpolated.go

```diff
@@ -17,36 +17,36 @@ type InterpolatedAssetPropertyValue struct {
 }
 
 func (p InterpolatedAssetPropertyValue) Frames(ctx context.Context, resources resource.ResourceProvider) (data.Frames, error) {
-	length := len(p.InterpolatedAssetPropertyValues)
 	property, err := resources.Property(ctx)
 	if err != nil {
 		return nil, err
 	}
+
 	// TODO: make this work with the API instead of ad-hoc dataType inference
 	// https://github.com/grafana/iot-sitewise-datasource/issues/98#issuecomment-892947756
 	if *property.AssetProperty.DataType == *aws.String("?") {
 		property.AssetProperty.DataType = aws.String(getPropertyVariantValueType(p.InterpolatedAssetPropertyValues[0].Value))
 	}
 
-	timeField := fields.TimeField(length)
-	valueField := fields.PropertyValueFieldForQuery(p.Query, property, length)
+	timeField := fields.TimeField(0)
+	valueField := fields.PropertyValueFieldForQuery(p.Query, property, 0)
 
 	frame := data.NewFrame(*property.AssetName, timeField, valueField)
 
 	frame.Meta = &data.FrameMeta{
 		Custom: models.SitewiseCustomMeta{
 			NextToken:  aws.StringValue(p.NextToken),
-			Resolution: "RAW", //circular dep
+			Resolution: p.Query.Resolution,
 		},
 	}
 
-	for i, v := range p.InterpolatedAssetPropertyValues {
+	for _, v := range p.InterpolatedAssetPropertyValues {
 		value := getPropertyVariantValue(v.Value)
 		if value == nil {
 			continue
 		}
-		timeField.Set(i, getTime(v.Timestamp))
-		valueField.Set(i, value)
+		timeField.Append(getTime(v.Timestamp))
```

Comment:
@ryantxu i took another look at this because it was bothering me, and i'm not sure where i was logging last time  looked it this, but it seems to be on our end. the fields were initialized with a length and skipping them on line 46 when `value == nil` causes the bug we were seeing.